File: src/copy-fsm/StatePattern/Tokenizer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StatePattern
{
    internal class Tokenizer
    {
        public static string[] tokenize(string input)
        {
            string[] tokens = input.Split(' ');
            List<string> ret = new List<string>();

            foreach(string token in tokens)
            {
                if (!token.Contains(","))
                {
                    ret.Add(token);
                }
                else
                {
                    string[] splitted = token.Split(',');
                    foreach (string split in splitted)
                    {
                        ret.Add(split);
                    }
                }
            }

            Console.WriteLine("tokens:"+ ret);
            return ret.ToArray<string>();
        }
    }
}

File: src/js/src/Tokenizer.js
"use strict";
exports.__esModule = true;
var Tokenizer = /** @class */ (function () {
    function Tokenizer(input) {
        if (input != null && input !== "") {
            this.input = input;
        }
        else {
            this.input = "";
        }
    }
    Tokenizer.prototype.tokenize = function () {
        var tokens = this.input.split(" ");
        var ret = [];
        for (var i = 0; i < tokens.length; i++) {
            if (tokens[i].includes(",")) {
                var splitted = tokens[i].split(",");
                for (var j = 0; j < splitted.length; j++) {
                    ret.push(splitted[j]);
                }
            }
            else {
                ret.push(tokens[i]);
            }
        }
        console.log("tokens: " + ret);
        return ret;
    };
    return Tokenizer;
}());
exports.Tokenizer = Tokenizer;
